packages/devtools/lib/src/debugger/debugger.dart

```diff
@@ -1097,15 +1271,24 @@ class ScriptsView implements CoreElementView {
   @override
   CoreElement get element => _items;
 
-  Stream<ScriptRef> get onSelectionChanged => _items.onSelectionChanged;
+  Stream<html.KeyboardEvent> get onKeyDown {
+    return _items.onKeyDown;
+  }
 
-  Stream<void> get onScriptsChanged => _items.onItemsChanged;
+  Stream<ScriptRef> get onSelectionChanged {
+    return _items.onSelectionChanged;
+  }
+
+  Stream<void> get onScriptsChanged {
```

Comment:
we need to do something to avoid changing the script to an invalid blank page when a user starts typing a file name in the box. we should only change the name once a valid file is selected or the user presses enter. Also make sure pressing enter always navs to the first file in the list of selections.

Reply:
Fixed a bunch of little things to make the UI interaction, better I hope.
- first ESC didn't work until something selected should always cancel.
- don't change script file until CLICK on script name or ENTER pressed while filtering and navigating in list
- when typing chars in the textfield should always select/highlight first name in list